File: eCinema.Web.API/ecinemamobile/lib/models/Movies/movies.dart
import 'dart:convert';
import 'dart:ffi';
import 'package:ecinemamobile/models/Movies/genres.movies.dart';
import 'package:ecinemamobile/models/Movies/movies.actors.dart';
import 'package:ecinemamobile/models/Schedules/schedule.dart';
import 'package:json_annotation/json_annotation.dart';

import '../Schedules/schedule.movie.dart';
import 'movies.directors.dart';
import 'movies.producer.dart';
import 'movies.writers.dart';
part 'movies.g.dart';

@JsonSerializable()
class Movies {
  int? id;
  String? title;
  int? releaseYear;
  int? duration;
  String? country;
  String? poster;
  String? synopsis;
  List<GenresMovies> genresMovies = [];
  List<ActorsMovies>? actorsMovies = [];
  List<DirectorsMovies>? directorsMovies = [];
  List<WritersMovies>? writersMovies = [];
  List<ProducersMovies>? producersMovies = [];
  List<ScheduleMovie>? schedules = [];

  Movies(
      this.id,
      this.title,
      this.releaseYear,
      this.duration,
      this.country,
      this.poster,
      this.synopsis,
      this.genresMovies,
      this.actorsMovies,
      this.directorsMovies,
      this.writersMovies,
      this.producersMovies,
      [this.schedules]);

  factory Movies.fromJson(Map<String, dynamic> json) => _$MoviesFromJson(json);

  Map<String, dynamic> toJson() => _$MoviesToJson(this);
}

File: eCinema.Web.API/ClassLibrary1/Entities/Movies.cs
using eCInema.Data.Entities;
using eCInema.Models.Entities;
using eCInema.Models.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCInema.Models
{
    public class Movies
    {
        public int Id { get; set; }
        public string Title { get; set; } = String.Empty;
        public string Synopsis { get; set; }
        public int ReleaseYear { get; set; }
        public int  Duration { get; set; }
        public List<MoviesGenres>? MoviesGenres { get; set; }
        public string Country { get; set; } 
        public List<DirectorsMovies>? DirectorsMovies { get; set; }
        public List<ActorsMovies>? ActorsMovies { get; set; }
        public List<ProducerMovies> ProducersMovies { get; set; }
        public List<WritersMovies> WritersMovies { get; set; }
        public byte[]? Poster { get; set; }//promijeniti iz nullabla
        public List<Schedule>? Schedules { get; set; }
    }
}
